module-2/backend/week_one.md

```diff
@@ -5,30 +5,123 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 9. What's the purpose of ERB?
+
+  ERB stands for embedded-ruby. It allows us to do ruby API methods inside of our HTML. So, as long as the appropriate data prep & manipulation has occurred and the view has access to this data, the data can be manipulated and displayed inside the view's HTML.
+  
 10. Why do I need a development AND test database?
+ 
+ The development database and test database are separate because we work in different environments throughout the life cycle of the application. While we want the test suite to have enough data to substantiate a test, we don't necessarily want to fully seed the database when in the testing environment due to time and memory resource usage.
+
 11. What's responsive design?
+
+ Responsive design is designing with a mobile-first strategy that yields a result that is responsive to the changing browser size. As the browser size changes, the content on the page shifts to be readable.
+
 12. What is CRUD and why is it important?
+
+  CRUD stands for Create, Read, Update, Delete. It's important because it represents everything we can do to data and the base functionality our app should have.
+  
 13. What does HTTP stand for? 
+
+ HyperText Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+ Renders on page: ``` <%= %>```
+ 
+ Is evaluated but not rendered: ``` <% %>```
+ 
 15. What's an ORM?
+
+ ORM stands for Object Relational Mapper. (I'm a bit fuzzy on the concept, but will do my best to explain without the documentation). I think an ORM is something that converts / is a middle man for Ruby objects and rows/tables in the database. This means that an individual row in a table is representative of the attributes of a Ruby object but is not a Ruby object itself. The ORM allows me to retrieve that row (instance) and puts a wrapper on it so I can interact with it with some of the Ruby API.
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+ ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+ Create
+  * User can add a restaurant
+    * GET '/restaurants/new' (retrieves form)
+    * POST '/restaurants' (adds new restaurant to db, displays all data on index view)
+   
+ Read
+   * User can see all restaurants
+     * GET '/restaurants'
+   * User can see a single restaurant
+     * GET '/restaurants/:id' 
+ 
+ Update
+   * User can update an existing restaurant
+     * GET '/restaurants/:id/edit' (retrieves form)
+     * PUT '/restaurants/:id' (updates to db, displays new data on show view)
+ 
+ Delete
+   * User can delete a restaurant
+     * DELETE '/restaurants/:id' (deletes from db, displays index view)

```

Comment:
This looks good! If you want more practice with this (per your response to question 1), I would recommend sketching this out with different resources (books, cats, teachers, schools, horses, muffins, cookies, etc.) when you have a break.

One small note: the POST, PUT, and DELETE actions don't actually display a new view; they redirect to other actions that do. For example, your POST '/restaurants' route adds a new restaurant, but then in order to display an index it would actually send instructions to the browser to make another request for a GET '/restaurants'. That's fairly nuanced, but something to be aware of.

Also, the particular view that is rendered after a PUT/POST/DELETE isn't necessarily part of this rule. You've given examples of what would be good practice, but if we wanted to send a user to a restaurant show page after they create a new restaurant that wouldn't be bad either.